sagemaker-python-sdk/tensorflow_script_mode_quickstart/tensorflow_script_mode_quickstart.ipynb

```diff
@@ -0,0 +1,274 @@
+    "\n",
+    "- **/opt/ml/model** use this directory to save models, checkpoints, or any other data. Any data saved in this folder is saved in the S3 bucket defined for training. See [model data](https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-training-algo.html#your-algorithms-training-algo-envvariables) for more information.\n",
+    "\n",
+    "### Reading additional information from the container\n",
+    "\n",
+    "Often, a user script needs additional information from the container that is not available in ```hyperparameters```.\n",
+    "SageMaker containers write this information as **environment variables** that are available inside the script.\n",
+    "\n",
+    "For example, the example above can read information about the **training** channel provided in the training job request by adding the environment variable `SM_CHANNEL_TRAINING` as the default value for the `--data_dir` argument:\n",
+    "\n",
+    "```python\n",
+    "if __name__ == '__main__':\n",
+    "  parser = argparse.ArgumentParser()\n",
+    "  # reads input channels training and testing from the environment variables\n",
+    "  parser.add_argument('--data_dir', type=str, default=os.environ['SM_CHANNEL_TRAINING'])\n",
+    "```\n",
+    "\n",
+    "Script mode displays the list of available environment variables in the training logs. You can find the [entire list here](https://github.com/aws/sagemaker-containers/blob/master/README.md#environment-variables-full-specification)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Training in SageMaker"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After you test the training job locally, upload the dataset to an S3 bucket so SageMaker can access the data during training:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "\n",
+    "inputs = sagemaker.Session().upload_data(path='sherlock', key_prefix='datasets/sherlock')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The returned variable inputs above is a string with a s3 location which SageMaker Tranining has permissions\n",
+    "to read data from. **It has education purposes, requiring\n",
+    " more robust solutions for larger datasets:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To train in SageMaker:\n",
+    "- change the estimator argument **train_instance_type** to any SageMaker ml instance available for training.\n",
+    "- set the **training** channel to a S3 location.\n",
+    "\n",
+    "For example:"
```

Comment:
is this part meant to be run? if so, I'd take out "for example:" and just talk about what we (we = the user and the author) are doing in each cell. If it is just an example, then let's put it in markdown so that it's not run with a user clicks "Cell --> Run All"